lib/deep_link_helper.dart

```diff
@@ -0,0 +1,74 @@
+import 'package:firebase_dynamic_links/firebase_dynamic_links.dart';
+
+class DeepLink {
+  final String deepLinkParameter;
+  final Function(String) action;
+
+  DeepLink({
+    this.deepLinkParameter,
+    this.action,
+  });
+}
+
+class DeepLinkHelper {
+  final List<DeepLink> deepLinks;
+
+  DeepLinkHelper({
+    this.deepLinks,
+  });
+
+  void init() {
+    initDynamicLinks();
+  }
+
+  void initDynamicLinks() async {
+    final PendingDynamicLinkData data =
+        await FirebaseDynamicLinks.instance.getInitialLink();
+
+    Uri deepLink = data?.link;
+
+    if (data?.link != null) {
+      _parseDeepLink(deepLink);
+    }
+
+    FirebaseDynamicLinks.instance.onLink(
+      onSuccess: (PendingDynamicLinkData dynamicLink) async {
+        final Uri deepLink = dynamicLink?.link;
+
+        if (deepLink != null) {
+          _parseDeepLink(deepLink);
+        }
+      },
+      onError: (OnLinkErrorException e) async {
+        print('onLinkError');
+        print(e.message);
+      },
+    );
+  }
+
+  void _saveDynamicLink(){
+    //TODO Implement save to disk pending deep link
```

Comment:
maybe a note of the use case support this would enable.... TODO: we need to implement this to be able to support opening deep links on very first open (no account created).

Reply:
Done